app/models/spree/gateway/amazon.rb

```diff
@@ -168,5 +207,40 @@ def random_suffix
       length = 10
       SecureRandom.random_number(36 ** length).to_s(36).rjust(length, '0')
     end
+
+    # Allows simulating errors in sandbox mode if the *last* name of the
+    # shipping address is "SandboxSimulation" and the *first* name is one of:
+    #
+    #   InvalidPaymentMethodHard-<minutes> (-<minutes> is optional. between 1-240.)
+    #   InvalidPaymentMethodSoft-<minutes> (-<minutes> is optional. between 1-240.)
+    #   AmazonRejected
+    #   TransactionTimedOut
+    #   ExpiredUnused-<minutes> (-<minutes> is optional. between 1-60.)
+    #   AmazonClosed
+    #
+    # E.g. a full name like: "AmazonRejected SandboxSimulation"
+    #
+    # See https://payments.amazon.com/documentation/lpwa/201956480 for more
+    # details on Amazon Payments Sandbox Simulations.
+    def sandbox_authorize_simulation_string(order)
+      return nil if !preferred_test_mode
+      return nil if order.ship_address.lastname != 'SandboxSimulation'
+
+      reason, minutes = order.ship_address.firstname.to_s.split('-', 2)
+      # minutes is optional and is only used for some of the reason codes
+      minutes ||= '1'
+
+      case reason
+      when 'InvalidPaymentMethodHard' then %({"SandboxSimulation": {"State":"Declined", "ReasonCode":"InvalidPaymentMethod", "PaymentMethodUpdateTimeInMins":#{minutes}}})
```

Comment:
Hm, they are long strings that I copy-pasted from Amazon's docs. I'm with you on shooting for 80 chars but any other arrangement I tried seemed like they made the code harder to read and/or seemed excessively verbose just for the goal of 80 chars.  Any suggestions though?